src/Searchbar.vue

```diff
@@ -35,18 +40,52 @@ export default {
             });
           }
         });
-        if (!hasMatchingHeading) {
-          if (regex.test(title) || regex.test(keywords)) {
-            matches.push(entry);
-          }
-        }
       });
       return matches;
     },
   },
   filters: {
     highlight(value, phrase) {
-      return value.replace(new RegExp(`(${phrase})`, 'gi'), '<mark>$1</mark>');
+      function mergeOverlappingIntervals(intervals) {
+        if (intervals.length <= 1) {
+          return intervals;
+        }
+        const sortedIntervals = intervals.sort((a, b) => a.start - b.start);
+        const stack = [sortedIntervals[0]];
+        for (let i = 1; i < sortedIntervals.length; i += 1) {
+          const top = stack[stack.length - 1];
+          if (top.end < sortedIntervals[i].start) {
+            stack.push(sortedIntervals[i]);
+          } else if (top.end < sortedIntervals[i].end) {
+            top.end = sortedIntervals[i].end;
+          }
+        }
+        return stack;
+      }
+
+      function collectMatchPos() {
```

Comment:
`collectMatchInterval()` might sound better? Then it will make sense why callers might eventually call `mergeOverlappingIntervals()`.

Reply:
How about `getMatchIntervals` since it's different from the `collect` behaviour in MarkBind?
And move it before `mergeOverlappingIntervals` for alphabetical order.